### PR TITLE
fix: [CDS-81169]: Adds runtime/expression support for time/timezone field for harness approval

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

This PR adds runtime and expression support for time and timezone field for harness approval step.

#### Screenshots

NA

https://github.com/harness/harness-core-ui/assets/96409598/dbab9963-c949-400e-950c-f81171983925

https://github.com/harness/harness-core-ui/assets/96409598/0c018830-9e7d-479e-ab5f-c81cba802755

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by samarth-sharma31** [comment]
trigger messagecheck

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/helper.ts [hunk 1]
```diff
@@ -136,3 +137,20 @@ export const scheduleAutoApprovalValidationSchema = (
...
+  const currentTimeOffset = getTimeZoneOffsetInMinutes(currentTimezone)
+
+  const currentTimeBasedOnTimezone = moment.utc(currentDate.getTime()).utcOffset(currentTimeOffset).valueOf()
+
+  const formValueTimeBasedOnTimezone =
+    moment.utc(currentTime, DATE_PARSE_FORMAT).utcOffset(currentTimeOffset).valueOf() - currentTimeOffset * 60000
+
+  const minApprovalTime: number = currentTimeBasedOnTimezone + parseStringToTime('15m')
+
+  return Boolean(formValueTimeBasedOnTimezone < minApprovalTime)
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/helper.ts:155]: Is `Boolean()` needed here? `varA < varB` should return a boolean

**Review by rameshx — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/helper.ts (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/Approval/helper.ts]: Agreed, should not be needed. Updated it.
Thanks @rameshx

**Review by ishantmehtaharness — APPROVED** [review]
looks good samarth

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**